data/tutorials/language/0lg_10_polymorphic_variants.md

```diff
@@ -0,0 +1,713 @@
+
+The expression ``[< `Broccoli | `Fruit of string ]`` plays the role of a type. However, it does not represent a single type, it represents three different types.
+- The type that only has `` `Broccoli`` as an inhabitant, its translation into a simple variant is `type t0 = Broccoli`
+- The type that only has `` `Fruit`` as an inhabitant, its translation into a simple variant is `type t1 = Fruit of string`
+- Type type that has both `` `Broccoli`` and `` `Fruit`` inhabitants, its translation into a simple variant is `type t2 = Broccoli | Fruit of string`
+
+Note each of the above translations into simple variants is correct. However, entering them as-is into the environment would lead to constructor shadowing (unless type annotation is used at pattern or expression level, see section on [Shared Constructors](#Shared-Constructors)).
+
+This also illustrates the other striking feature of polymorphic variants: values can be attached to several types. For instance, the tag `` `Broccoli`` inhabits ``[ `Broccoli ]`` and``[ `Broccoli | `Fruit of String ]``, but it also inhabits any type that contains it.
+
+What is displayed by the type-checker, for instance ``[< `Broccoli | `Fruit of string ]``, isn't a single type. It is a type expression that designates a constrained set of types. For instance, all the types that are defined by a group of tags containing either`` `Broccoli`` or `` `Fruit of string`` and nothing more. This is the meaning of the `<` sign in this type expression. This is a bit similar to what happens with `'a list`, which isn't a single type either, but a type expression that designates the set of `list` types of something, i.e. the set of types where `'a` has been replaced by some other type. It is also meant to indicate that the exact types are subsets of the indicated ones (this will be explained in the section on [Subtyping](#Subtyping-of-Polymorphic-Variants)).
+
+This is the sense of the polymorphism of polymorphic variants. Polymorphic variants types are type expressions. The structural typing algorithm used for polymorphic variants creates type expressions that designate sets of types (here the three types above), which are defined by constraints on sets of tags (the inequality symbols). The polymorphism of polymorphic variants is different.
+
+In the rest of this tutorial, the following terminology is used:
+- “simple variants”: product types and variants such as `list`, `option`
+- polymorphic variant: type expressions displayed by the OCaml type-checker such as ``[< `Broccoli | `Fruit of string ]``
+
+### Tag Sharing
+
+Here is another function using pattern matching on a polymorphic variant.
+```ocaml
+# let g = function `Edible name -> name | `Broccoli -> "Brassica oleracea";;
+val g : [< `Broccoli | `Edible of string ] -> string = <fun>
+
+# f `Broccoli;;
```

Comment:
Consider that if I jump into this section, I don't know where `f` came from or what it is. Keeping it self-contained should be okay since these examples are super small.

Reply:
You are right. The document does not state explicitly examples are meant to be played in order and jumping into a section does not work. This needs to be clarified.